File: client/src/app/models/account-models/Follower.ts
import {Account} from "./Account";

export class Follower {

  private id: number;
  private followerUsername: string;
  private account: Account;

  constructor() { }


  getId(): number {
    return this.id;
  }

  setId(value: number) {
    this.id = value;
  }

  getFollowerUsername(): string {
    return this.followerUsername;
  }

  setFollowerUsername(value: string) {
    this.followerUsername = value;
  }

  getAccount(): Account {
    return this.account;
  }

  setAccount(value: Account) {
    this.account = value;
  }

  static anyToObject(any: any){
    let account = Account.anyToObject(any.account);
    let follower = new Follower();
    follower.setFollowerUsername(any.followerUsername);
    follower.setAccount(any.account);
    return follower;
  }

}

File: src/main/java/com/app/demo/entities/accountEntities/Follower.java
package com.app.demo.entities.accountEntities;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
public class Follower {

    @Id
    @NotNull
    @GeneratedValue
    @Column(name = "id", nullable = false, unique = true)
    private Long id;

    @NotNull
    @Column(name = "followerUsername")
    private String followerUsername;

    @NotNull
    @ManyToOne
    private Account account;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getFollowerUsername() {
        return followerUsername;
    }

    public void setFollowerUsername(String followerUsername) {
        this.followerUsername = followerUsername;
    }

    public Account getAccount() {
        return account;
    }

    public void setAccount(Account account) {
        this.account = account;
    }

    @Override
    public String toString() {
        return "Follower{" +
                ", followerUsername='" + followerUsername + '\'' +
                ", account=" + account +
                '}';
    }
}